packages/devtools_app/lib/src/debugger/evaluate.dart

```diff
@@ -274,55 +284,232 @@ class _ExpressionEvalFieldState extends State<ExpressionEvalField>
   }
 }
 
+// If Dart had union types, ref would be type ClassRef | FuncRef | LibraryRef
+Future<LibraryRef> findOwnerLibrary(
+    Object ref, DebuggerController controller) async {
+  if (ref is LibraryRef) {
+    return ref;
+  }
+  if (ref is ClassRef) {
+    final clazz = await classFor(ref, controller);
+    return clazz?.library;
+  }
+  if (ref is FuncRef) {
+    return findOwnerLibrary(ref.owner, controller);
+  }
+  return null;
+}
+
 Future<List<String>> autoCompleteResultsFor(
   EditingParts parts,
   DebuggerController controller,
 ) async {
   final result = <String>{};
   if (!parts.isField) {
-    result.addAll(
-        controller.variables.value.map((variable) => variable.boundVar.name));
+    final variables = controller.variables.value;
```

Comment:
do we need to use List.from here if the intent is to make a copy?

Reply:
We're not making a copy. We are just storing as a local variable two places within this method.